wrappers/services_test.go

```diff
@@ -3151,6 +3183,15 @@ apps:
 	info := snaptest.MockSnap(c, surviveYaml, &snap.SideInfo{Revision: snap.R(1)})
 	srvFile := "snap.test-snap.foo.service"
 
+	r := systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		if out := mockSystemCtlShow(cmd, nil); out != nil {
+			return out, nil
+		}
+		s.sysdLog = append(s.sysdLog, cmd)
```

Comment:
Same suggestion to rearrange the calls:

```suggestion
		s.sysdLog = append(s.sysdLog, cmd)
		if out := mockSystemCtlShow(cmd, nil); out != nil {
			return out, nil
		}
```

This will probably need an update in unit tests.

Reply:
Please have a look now, the last commit addresses this. However, I did prefer the previous implementation, because this change makes the tests' code less clear. I would actually go even further, and remove all calls to `systemctl show` from the logging, so that we test only those actions that produce a state change.